synapse/handlers/room.py

```diff
@@ -732,6 +738,15 @@ async def create_room(
         invite_3pid_list = config.get("invite_3pid", [])
         invite_list = config.get("invite", [])
 
+        # validate each entry for correctness
+        for invite_3pid in invite_3pid_list:
+            if not self._has_all_3pid_keys(invite_3pid):
+                raise SynapseError(
+                    400,
+                    f"`id_server` and `id_access_token` are required when doing 3pid invite, caused by {invite_3pid}",
```

Comment:
Suppose we have an `invite_3pid` with `id_server` and `id_access_token` but no `medium`. We will still log this error. That seems incorrect.

Reply:
So if I understand correctly, medium is always optional and does not have to be in the invite? Can it be ommited or should another error be raised?

Reply:
Although I think it would just have to log "all of the (4) fields need to be present in an invite, caused by:" if the if above it is correct

Reply:
medium is not optional.

Reply:
Okay, so in newest version all these properties are mentioned as mandatory in error message.